Library/Homebrew/formulary.rb

```diff
@@ -118,6 +121,7 @@ def initialize(alias_path)
       path = alias_path.resolved_path
       name = path.basename(".rb").to_s
       super name, path
+      @install_name = alias_path.basename.to_s
```

Comment:
I was afraid to try. :P

I imagine `Formula#name` is used pretty widely across Homebrew. This would, for example, change the install prefix for a formula based on the alias used to install it, which doesn't seem like a great idea to me.